src/lib/rate-limiter-valkey.ts

```diff
@@ -0,0 +1,69 @@
+/**
```

Comment:
Nit: but perhaps we should drop "valkey" from the filename or name it "redis-compat" or something.